master/buildbot/test/unit/test_process_buildstep.py

```diff
@@ -200,6 +200,87 @@ def test_regularLocks(self):
         self.expectOutcome(result=SUCCESS)
         yield self.runStep()
 
+    @defer.inlineCallbacks
```

Comment:
Yes. Without my changes to buildstep.py, this assert will trigger:
https://github.com/buildbot/buildbot/blob/3e483d80468eee39012cc1381305b3edd18f16bd/master/buildbot/test/unit/test_process_buildstep.py#L263
At this point we have cancelled stepc, so why is lock1 available for this step? To prove further that it is something actually wrong here, I commented out all the asserts for my test. My test will still not pass, as it will become stuck at https://github.com/buildbot/buildbot/blob/3e483d80468eee39012cc1381305b3edd18f16bd/master/buildbot/test/unit/test_process_buildstep.py#L278
meaning that `stepd` never manages to `acquireLocks` (even though both locks are now available). With the changes to buildstep.py, it obviously passes.